src/main/java/net/floodlightcontroller/perfmon/CumulativeTimeBucket.java

```diff
@@ -69,13 +69,16 @@ public int getNumComps() {
         return compStats.values();
     }
 
-    public CumulativeTimeBucket(List<IOFMessageListener> listeners) {
-        compStats = new ConcurrentHashMap<Integer, OneComponentTime>(listeners.size());
-        for (IOFMessageListener l : listeners) {
-            OneComponentTime oct = new OneComponentTime(l);
+    public CumulativeTimeBucket() {
+        compStats = new ConcurrentHashMap<>();
+        startTime_ns = System.nanoTime();
+    }
+
+    public void addListener(IOFMessageListener listener) {
+        if (!compStats.containsKey(listener.hashCode())) {
```

Comment:
I haven’t looked at the entire module, but a hashmap with hash codes as the keys sounds odd. You typically have a object as the key. I might just not know enough about your implementation though

Reply:
Thanks! I agree it is little odd -- this "compStats" hashmap structure wasn't my implementation.  As I started this patch I focused on add IOFMessageListener separately rather than passing them all into the CumulativeTimeBucket constructor(otherwise every time the bootstrap method being called there will be a new CumulativeTimeBucket initiated), so I left this hashmap structure as it is.

Reply:
@rizard Please let me know your thoughts, thanks!